contracts/tokenbridge/arbitrum/L2AtomicTokenBridgeFactory.sol

```diff
@@ -229,6 +241,10 @@ contract L2AtomicTokenBridgeFactory {
             ITransparentUpgradeableProxy(canonicalL2WethGateway), l2WethGatewayLogic
         );
 
+        // init logic contract with dummy values
+        address dead = address(0x000000000000000000000000000000000000dEaD);
+        L2WethGateway(payable(l2WethGatewayLogic)).initialize(dead, dead, dead, dead);
```

Comment:
Does l2WethLogic also need init? maybe we should include a comment when we dont init logic contracts as to why

Reply:
It does not need init because it is set to `initialized` state at creation time here - https://github.com/OffchainLabs/token-bridge-contracts/blob/127c1f3d301a173a64dc27040df394a10938dbb3/contracts/tokenbridge/libraries/aeERC20.sol#L30

Added comments here - https://github.com/OffchainLabs/token-bridge-contracts/pull/54/commits/127c1f3d301a173a64dc27040df394a10938dbb3

Reply:
https://github.com/OffchainLabs/token-bridge-contracts/pull/55